From f65c14c80c779fbd9f5dd3e18c1e883c4efe663d Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 13 Mar 2025 13:06:46 +0200
Subject: [PATCH] [FSW-13187] fs1: mem_copy: skip barrier for sim

coral has issue with barrier (inflights bit is
not set after transactions are done) SW-222241
Hence, as a temporary WA, it is required
to comment out the barrier, otherwise, fit load and
other process will fail in coral.

Change-Id: Ic2f420db9fa196467917c19d348605c3d08ea70a
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h b/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
index 2e43338..2ce9ff9 100644
--- a/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
+++ b/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
@@ -176,16 +176,19 @@
 	u32 __val, __addr;									\
 	int err;										\
 												\
-	__addr = mmIO_0_CFG_HALF_0_PSOC_ARC_MNG_LOCAL_BASE +					\
-		 mmARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2;					\
-	err = readl_poll_busy_wait_timeout(__addr,						\
-			__val,									\
-			__val & ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_NO_WR_INFLIGHT_M,	\
-			HL_W_POLL_DELAY_US,							\
-			HL_W_BARRIER_TIMEOUT_US);						\
-	if (err)										\
-		LOG_ERR("HL_MNG_W_BARRIER timeout");						\
-												\
+	/* Temporarily comment-out barrier for coral till issue SW-222241 is resolved */	\
+	if (!hl_is_sim_mode()) {								\
+		__addr = mmIO_0_CFG_HALF_0_PSOC_ARC_MNG_LOCAL_BASE +				\
+			mmARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2;				\
+		err = readl_poll_busy_wait_timeout(__addr,					\
+				__val,								\
+				__val & ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_NO_WR_INFLIGHT_M,\
+				HL_W_POLL_DELAY_US,						\
+				HL_W_BARRIER_TIMEOUT_US);					\
+		if (err)									\
+			LOG_ERR("HL_MNG_W_BARRIER timeout");					\
+	}											\
+done:												\
 	err;											\
 })
 
